### PR TITLE
fix(ci,release): run on any new tag or trigger manually

### PR DESCRIPTION
**Pull request by cristaloleg**
Rework of #3834

### CODE REVIEW

File: .github/workflows/ci_release.yml [hunk 1]
```diff
@@ -1,15 +1,18 @@
 name: CI and Release
+
 on:
   merge_group:
   push:
     branches:
       - main
+    tags:
+      - '**'
```
Comment (cristaloleg) [inline .github/workflows/ci_release.yml:9]: Trigger on any pushed tag.

File: .github/workflows/ci_release.yml [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
   release:
     types: [published]
   pull_request:
   workflow_dispatch:
     inputs:
-      version:
+      version-bump:
```
Comment (cristaloleg) [inline .github/workflows/ci_release.yml:15]: Rename 'cause it's not a version but version bump (Captain Obvious 🫡)

File: .github/workflows/ci_release.yml [hunk 3]
```diff
@@ -19,6 +22,9 @@ on:
           - patch
           - minor
           - major
+      tag-as:
```
Comment (cristaloleg) [inline .github/workflows/ci_release.yml:25]: Input for manual trigger. Basically this case:
image

File: .goreleaser.yaml [hunk 4]
```diff
@@ -1,3 +1,4 @@
+version: 2
```
Comment (cristaloleg) [inline .goreleaser.yaml:1]: `goreleaser` errored without an explicit config version

File: .github/workflows/ci_release.yml [hunk 5]
```diff
@@ -121,6 +127,13 @@ jobs:
         with:
           go-version: ${{ needs.setup.outputs.go-version }}
 
+      - name: Bump version and push tag
+        if: ${{ github.event_name == 'workflow_dispatch' }} && ${{ inputs.tag-as }} != ''
```
Comment (cristaloleg) [inline .github/workflows/ci_release.yml:131]: Do a tag on manual trigger with a specified tag. `goreleaser --snapshot` (or `--auto-snapshot`) doesn't publish artifacts 🙄 https://github.com/goreleaser/goreleaser/issues/1538#issuecomment-927124677

**Review by renaynay — APPROVED** [review]
utack , thank you for testing this separately @cristaloleg

**Review by Wondertan — APPROVED** [review]
utack

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**